explorer/mempool.go

```diff
@@ -0,0 +1,228 @@
+			exp.MempoolData.NumRegular++
+		case "Revocation":
+			exp.MempoolData.Revocations = append([]MempoolTx{tx}, exp.MempoolData.Revocations...)
+			exp.MempoolData.NumRevokes++
+		}
+
+		exp.MempoolData.LatestTransactions = append([]MempoolTx{tx}, exp.MempoolData.LatestTransactions[:len(exp.MempoolData.LatestTransactions)-1]...)
+
+		exp.MempoolData.NumAll++
+		exp.MempoolData.TotalOut += tx.TotalOut
+		exp.MempoolData.TotalSize += tx.Size
+		exp.MempoolData.FormattedTotalSize = humanize.Bytes(uint64(exp.MempoolData.TotalSize))
+
+		exp.MempoolData.Unlock()
+		exp.wsHub.HubRelay <- sigNewTx
+		exp.wsHub.NewTxChan <- &tx
+	}
+}
+
+func (exp *explorerUI) StopMempoolMonitor(txChan chan *NewMempoolTx) {
+	log.Infof("Stopping mempool monitor")
+	txChan <- nil
+}
+
+func (exp *explorerUI) StartMempoolMonitor(newTxChan chan *NewMempoolTx) {
+	go exp.mempoolMonitor(newTxChan)
+}
+
+func (exp *explorerUI) storeMempoolInfo() {
+
+	defer func(start time.Time) {
+		log.Debugf("storeMempoolInfo() completed in %v",
+			time.Since(start))
+	}(time.Now())
+
+	exp.MempoolData.Lock()
+	defer exp.MempoolData.Unlock()
+
+	memtxs := exp.blockData.GetMempool()
+
+	if memtxs == nil {
+		log.Error("Could not get mempool transactions")
+		return
+	}
+	sort.Sort(byTime(memtxs))
+
+	var latest []MempoolTx
+	if len(memtxs) > 5 {
+		latest = memtxs[:5]
+	} else {
+		latest = memtxs
+	}
+
+	tickets := make([]MempoolTx, 0)
+	votes := make([]MempoolTx, 0)
+	revs := make([]MempoolTx, 0)
+	regular := make([]MempoolTx, 0)
+
+	var totalOut float64
+	var totalSize int32
+
+	for _, tx := range memtxs {
+		switch tx.Type {
+		case "Ticket":
+			tickets = append(tickets, tx)
+		case "Vote":
+			votes = append(votes, tx)
+		case "Revocation":
+			revs = append(revs, tx)
+		default:
+			regular = append(regular, tx)
+		}
+		totalOut += tx.TotalOut
+		totalSize += tx.Size
+	}
+
+	exp.NewBlockDataMtx.Lock()
+	lastBlock := exp.NewBlockData.Height
+	lastBlockTime := exp.NewBlockData.BlockTime
+	exp.NewBlockDataMtx.Unlock()
+
+	exp.MempoolData.Transactions = regular
+	exp.MempoolData.Tickets = tickets
+	exp.MempoolData.Revocations = revs
+	exp.MempoolData.Votes = votes
+
+	exp.MempoolData.MempoolShort = MempoolShort{
+		LastBlockHeight:    lastBlock,
+		LastBlockTime:      lastBlockTime,
+		TotalOut:           totalOut,
+		TotalSize:          totalSize,
+		NumAll:             len(memtxs),
+		NumTickets:         len(tickets),
+		NumVotes:           len(votes),
+		NumRegular:         len(regular),
+		NumRevokes:         len(revs),
+		LatestTransactions: latest,
+		FormattedTotalSize: humanize.Bytes(uint64(totalSize)),
+	}
+}
+
+type byTime []MempoolTx
+
+func (txs byTime) Less(i, j int) bool {
+	return txs[i].Time > txs[j].Time
+}
+
+func (txs byTime) Len() int {
+	return len(txs)
+}
+
+func (txs byTime) Swap(i, j int) {
+	txs[i], txs[j] = txs[j], txs[i]
+}
+
+type byType []MempoolTx
+
+func (txs byType) Less(i, j int) bool {
+	weights := map[string]int{
```

Comment:
this gets constructed every time `Less` is run.  Lets make it a global or deal with it some other way.

Reply:
for instance, we could ditch the `sort.Interface` approach used her, and use[ `sort.Slice`](https://golang.org/pkg/sort/#Slice) where the specified closure (`less`) would capture the map if it were defined as a meber of `explorerUI`.  Just a though

Reply:
Removed `ByType` entirely, it is no longer needed since the transactions are already sorted into their respective slices but I'll keep in mind the `sort.Slice` approach.